src/connection.rs

```diff
@@ -306,6 +313,48 @@ impl<S, C> DeviceConnection<S, C> {
         Ok(true)
     }
 
+    async fn remove_interfaces<'a, I>(&mut self, interfaces_name: I) -> Result<Vec<String>, Error>
+    where
+        C: Register,
+        S: PropertyStore,
+        I: IntoIterator<Item = &'a str> + Clone + Send,
+    {
+        let mut interfaces = self.interfaces.write().await;
+
+        let to_remove = interfaces_name
+            .clone()
+            .into_iter()
+            .filter_map(|iface_name| {
+                let interface = interfaces.get(iface_name).map(|i| (i.interface_name(), i));
+
+                // only in debug mode, if the interface is not found, it has been probably misspelled
+                debug_assert!(interface.is_some(), "interface {iface_name} not found");
+                if interface.is_none() {
+                    warn!("{iface_name} not found, skipping");
+                }
+
+                interface
+            })
+            .collect();
+
+        self.connection
+            .remove_interfaces(&interfaces, &to_remove)
+            .await?;
+
+        for (iface_name, _) in to_remove.iter() {
+            // We cannot error here since we already unsubscribed to the interface
+            if let Err(err) = self.store.delete_interface(iface_name).await {
+                error!("failed to remove property {err}");
+            }
+        }
+
+        let removed_names = to_remove.keys().map(|k| k.to_string()).collect_vec();
+
+        interfaces.remove_many(&HashSet::from_iter(interfaces_name));
```

Comment:
With the changes on the remove many of interfaces, this can be simplified in `to_remove.keys()` without allocating the hashset

Reply:
It's `&HashSet::from_iter(interfaces_name)` that changes, not `removed_names`, but I got your point. Thanks